feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapMainSettingsViewModel.kt

```diff
@@ -109,6 +113,10 @@ class SwapMainSettingsViewModel(
         emptyAssetTitle = R.string.swap_field_asset_to_title
     )
 
+    val amountOutFiat = combine(quotingState, amountOutInput.fiatAmount) { quote, fiat ->
+        formatOutFiat(quote, fiat)
+    }
```

Comment:
I think we can make `SwapAmountInputMixin` aware of `priceImpact` and perform formatting internally using `PriceImpactFormatter`. We can achieve that by optionally passing `priceImpactFlow: Flow<Percent?>` to the mixin factory 
It will save us from making such kind of add-ons every time we need a price impact formatting